lib/circuitbox/memory_store/compactor.rb

```diff
@@ -0,0 +1,33 @@
+# frozen-string-literal: true
+
+require_relative 'monotonic_time'
+
+class Circuitbox
+  class MemoryStore
+    class Compactor
+      include MonotonicTime
+
+      attr_reader :store, :frequency, :compact_after
+
+      def initialize(store:, frequency:)
+        @store = store
+        @frequency = frequency
+        set_next_compaction_time
+      end
+
+      def run
+        return unless compact_after < current_second
+
+        @store.delete_if { |_, value| value.expired? }
```

Comment:
With all the locking we're doing around the shared data structure I'm a little concerned about this becoming a bottleneck. I think the compaction frequency is a good way to prevent this from running too often. We could also keep track of an earliest expired time so that we don't run the compaction when there are no keys that have expired.

Reply:
I was going to look at increasing the default compaction_frequency so this happens less often. I may have another optimization I can make here to reduce the runtime of a compaction.